samza-rest/src/main/java/org/apache/samza/rest/script/ScriptRunner.java

```diff
@@ -126,4 +153,34 @@ public void run() {
     log.debug("Exit value {}", exitVal);
     return exitVal;
   }
+
+  /**
+   * @return true if this runner will forward the current environment variables to the child process, false otherwise.
+   */
+  public boolean forwardEnvironmentVars() {
+    return forwardEnvironmentVars;
+  }
+
+  /**
+   * Set the flag indicating whether this runner will forward the current environment variables to the child process.
+   *
+   * @param forwardEnvironmentVars  true to forward the current environment to the child process,
+   *                                false to start with an empty environment.
+   */
+  public void setForwardEnvironmentVars(boolean forwardEnvironmentVars) {
```

Comment:
Passing it at construction isn't doable without significant work, because the ScriptJob proxy reuses the same runner over and over. 

The second approach sounds promising. It gave me another idea. 
I expect it to be rare that we would want to forward the env vars. Maybe I'll just clear them always and then the caller can use .environment() to get the map and copy the current env into it, if they really want to forward. 

Let me try that and post another diff. 

Thanks for the suggestion!